Fix build for circt 2500cf570bbc6c6d45940466f9421e93d3a49132

tshadley opened the pull request:
This was in reference to https://github.com/circt-hls/circt-hls/issues/11

All tests pass except for cosim insertion_sort (hanging in mlir-cpu-runner):

   "Timed Out Test: CIRCT-HLS cosim :: suites/Dynamatic/insertion_sort/tst_insertion_sort.c"

I also added a bash build script "build.sh" for Ubuntu 22.04.1 LTS

Let me know if this is helpful or if further steps/updates should be done.  Also open to debugging insertion_sort.c further if you have any hints.

Best regards,
-Tedd

mortbopet commented on cosim_test/lit.cfg.py:
```diff
@@ -62,7 +62,8 @@
 # Enable Polygeist if it has been detected.
 if config.polygeist_bin_dir != "":
-  tool_dirs.append(os.path.dirname(config.polygeist_bin_dir))
+  #tool_dirs.append(os.path.dirname(config.polygeist_bin_dir))
```
please remove comment

mortbopet commented on include/circt-hls/Dialect/Cosim/CMakeLists.txt:
```diff
@@ -3,8 +3,10 @@ add_mlir_doc(Cosim -gen-dialect-doc Cosim Dialects/)
 
-mlir_tablegen(CosimAttrs.h.inc -gen-struct-attr-decls)
-mlir_tablegen(CosimAttrs.cpp.inc -gen-struct-attr-defs)
+#mlir_tablegen(CosimAttrs.h.inc -gen-struct-attr-decls)
```
please remove comments

mortbopet commented on include/circt-hls/InitAllDialects.h:
```diff
@@ -25,7 +25,7 @@
 #include "circt/Dialect/SV/SVDialect.h"
 #include "circt/Dialect/Seq/SeqDialect.h"
-#include "circt/Dialect/StaticLogic/StaticLogic.h"
+//#include "circt/Dialect/StaticLogic/StaticLogic.h"
```
please remove comment

mortbopet commented on tools/hls-opt/CMakeLists.txt:
```diff
@@ -38,19 +38,23 @@ target_link_libraries(hls-opt
   CIRCTStandardToHandshake
-  CIRCTStandardToStaticLogic
-  CIRCTStaticLogicOps
+  #CIRCTStandardToStaticLogic
+  #CIRCTStaticLogicOps
```
please remove comments in this file

mortbopet commented on tools/hls-opt/hls-opt.cpp:
```diff
@@ -11,19 +11,20 @@
-#include "mlir/Dialect/SCF/SCF.h"
-#include "mlir/Pass/PassRegistry.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+//#include "mlir/Pass/PassRegistry.h"
```
please remove comment

mortbopet reviewed (commented):
Thank you very very much for this! Regular bumps are important for this repo, so glad to have help with this.
wrt. the cosim test that is failing - no worries, it's probably due to some upstream logical changes. It's more important that this repo gets up and running.

To improve things even further, would you mind:
- adding a `circt_tag.txt` file in the base of this repository which points to the latest CIRCT version that we're tracking (i,e. the commit hash of your bump).
- modify `README.md` to describe this version as our "known good" version (at the place where (after [this](https://github.com/circt-hls/circt-hls#build-circt)).
- For a+ effort, modify our CI builds such that they will load this tag file and use that as the commit hash for the CIRCT checkout

tshadley commented:
Glad to help! I also appreciate your guiding me through the process.

I bumped CIRCT to 062eab31a67c8a8418b2ba9fb4f1814cdca08783 with no issues.

> adding a circt_tag.txt file 

Added this file.

> modify README.md to describe this version as our "known good" version (at the place where (after this).

Done.  I also added a line about `build.sh`, free free to omit if preferred (I find I prefer the bash flow to anything else at the moment).

> For a+ effort, modify our CI builds such that they will load this tag file and use that as the commit hash for the CIRCT checkout

I used `juliangruber/read-file-action@v1` to read the tag file into a variable and specified that as CIRCT `ref`.  I tested with `act` (which runs github actions locally), but this might not be the same as a true github flow.  

I cleaned up the unneccessary comments.

tshadley commented:
I originally ran into the apt-get 404 errors using the  `act` local workflow simulator.  These were fixed by calling
`apt-get update --fix-missing` 

I assumed these were due to some eccentricities of `act` but I see you're getting the same error now using github workflow?  This might be due to the docker image being slightly out of date now.   In any case, fix-missing seems to solve it on my flow (78febfc)
